crates/ruff_python_formatter/src/comments/map.rs

```diff
@@ -539,11 +539,11 @@ struct PartIndex(NonZeroU32);
 impl PartIndex {
     fn from_len(value: usize) -> Self {
         assert!(value < u32::MAX as usize);
-        // SAFETY:
+        // OK because:
         // * The `value < u32::MAX` guarantees that the add doesn't overflow.
         // * The `+ 1` guarantees that the index is not zero
-        #[allow(unsafe_code, clippy::cast_possible_truncation)]
-        Self(unsafe { std::num::NonZeroU32::new_unchecked((value as u32) + 1) })
+        #[allow(clippy::cast_possible_truncation)]
+        Self(std::num::NonZeroU32::new((value as u32) + 1).expect("valid value"))
```

Comment:
I'm surprised there isn't a method in std that combines the assert and `NonZeroU32::new` (like `NonZeroU32::try_from(value + 1)`, but apparently `NonZeroU32` doesn't even have a "normal" add

Reply:
In this case, it would need to be a `TryFrom<usize> for NonZeroU32`. And if you fill out the full complement of those across all the `NonZero` types, it ends up being quite a lot of impls. Not clear that's worth doing when you can compose things, e.g., `u32::try_from(value).ok().and_then(std::num::NonZeroU32::new)` or something like that.

And yeah it doesn't have `Add` because there are some thorny decisions around its semantics. e.g., What happens when you do `NonZeroU32::MAX + 1`. And if you had subtraction, then `NonZeroU32::new(1).unwrap() - 1`. You'd have to panic I think.

I also don't think anyone has seriously proposed adding arithmetic methods to the various `NonZero` types either.